gateway/reverse_proxy.go

```diff
@@ -910,16 +918,17 @@ func (p *ReverseProxy) CopyResponse(dst io.Writer, src io.Reader) {
 		}
 	}
 
-	p.copyBuffer(dst, src, nil)
+	p.copyBuffer(dst, src)
 }
 
-func (p *ReverseProxy) copyBuffer(dst io.Writer, src io.Reader, buf []byte) (int64, error) {
-	if len(buf) == 0 {
-		buf = make([]byte, 32*1024)
-	}
+func (p *ReverseProxy) copyBuffer(dst io.Writer, src io.Reader) (int64, error) {
```

Comment:
I think it is worth to support this change with new benchmark vs master (would have to add it to master manually to have numbers `"before"`)